src/ports/postgres/modules/kmeans/kmeans_auto.py_in

```diff
@@ -0,0 +1,201 @@
+    min_frac_reassigned=None, k_selection_algorithm=None, seeding=None,
+    seeding_sample_ratio=None, **kwargs):
+
+    _validate(output_table, k)
+
+    (fn_dist, agg_centroid, max_num_iterations, min_frac_reassigned,
+     k_selection_algorithm, seeding_sample_ratio) = set_defaults(
+        schema_madlib, fn_dist, agg_centroid, max_num_iterations,
+        min_frac_reassigned, k_selection_algorithm, seeding,
+        seeding_sample_ratio)
+
+    silhouette = ""
+    elbow = ""
+
+    plpy.execute("""
+        CREATE TABLE {output_table} (
+            k INTEGER,
+            centroids   DOUBLE PRECISION[][],
+            cluster_variance    DOUBLE PRECISION[],
+            objective_fn    DOUBLE PRECISION,
+            frac_reassigned DOUBLE PRECISION,
+            num_iterations  INTEGER)
+        """.format(**locals()))
+
+    silhouette_vals = []
+
+    for current_k in k:
+        if seeding is 'random':
+            plpy.execute("""
+                INSERT INTO {output_table}
+                SELECT {current_k} as k, *
+                FROM {schema_madlib}.kmeans_random('{rel_source}',
+                                     '{expr_point}',
+                                     {current_k},
+                                     '{fn_dist}',
+                                     '{agg_centroid}',
+                                     {max_num_iterations},
+                                     {min_frac_reassigned});
+                """.format(**locals()))
+        else:
+            plpy.execute("""
+                INSERT INTO {output_table}
+                SELECT {current_k} as k, *
+                FROM {schema_madlib}.kmeanspp('{rel_source}',
+                                     '{expr_point}',
+                                     {current_k},
+                                     '{fn_dist}',
+                                     '{agg_centroid}',
+                                     {max_num_iterations},
+                                     {min_frac_reassigned},
+                                     {seeding_sample_ratio});
+                """.format(**locals()))
+
+        if k_selection_algorithm != 'elbow':
+            silhouette_query= """
+                SELECT * FROM {schema_madlib}.simple_silhouette(
+                    '{rel_source}',
+                    '{expr_point}',
+                    (SELECT centroids
+                     FROM {output_table}
+                     WHERE k = {current_k}),
+                    '{fn_dist}')
+                """.format(**locals())
+            silhouette_vals.append(
+                plpy.execute(silhouette_query)[0]['simple_silhouette'])
+
+    # If the selection is silhouette or both, calculate silhouette
+    if k_selection_algorithm != 'elbow':
+
+        silhouette_vals_np = np.array(silhouette_vals)
+        optimal_sil =  k[np.argmax(silhouette_vals_np)]
+        silhouette = ", {0} AS silhouette".format(max(silhouette_vals))
```

Comment:
I would recommend moving the `as silhouette` statement to the create table command. Then we can also use the global variable `SILHOUETTE` instead of the hard coded value.
same goes for elbow